File: fds-dashboard/src/packages/user-management-service/src/auth/request/reset-password-request.ts
import * as z from "zod";

export type ResetPasswordRequest = {
  password: string;
  confirmPassword: string;
  token: string;
};

export const ResetPasswordSchema = z.object({
  password: z.string(),
  confirmPassword: z.string(),
  token: z.string(),
});
export type ResetPasswordZod = z.infer<typeof ResetPasswordSchema>;

File: fds-backend/user-management-service/src/main/java/com/yokke/usermanagement/auth/request/ResetPasswordRequest.java
package com.yokke.usermanagement.auth.request;

import com.yokke.usermanagement.auth.validation.NoWhitespace;
import com.yokke.usermanagement.auth.validation.PasswordMatches;
import com.yokke.usermanagement.auth.validation.ValidPassword;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.*;

@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@PasswordMatches

public class ResetPasswordRequest {
    @ValidPassword
    @NoWhitespace
    @NotBlank
    @NotNull
    private String password;
    @ValidPassword
    @NoWhitespace
    @NotBlank
    @NotNull
    private String confirmPassword;
    private String token;

}
